### PR TITLE
Adding https://meetfranz.com/

### PR DESCRIPTION
**Pull request by vbatts**
I put an email in the appdata that can be dropped or whatever.

Signed-off-by: Vincent Batts <vbatts@hashbangbash.com>

### CODE REVIEW

File: com.meetfranz.Franz.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.meetfranz.Franz.desktop</id>
```
Comment (bilelmoussaoui) [inline com.meetfranz.Franz.appdata.xml]:
```suggestion
  <id>com.meetfranz.Franz</id>
```

File: com.meetfranz.Franz.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  <name>Franz</name>
+  <project_license>Apache-2.0</project_license>
+  <developer_name>Stefan Maizner</developer_name>
+  <summary>Messenger for the desktop</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <url type="homepage">https://meetfranz.com/</url>
```
Comment (bilelmoussaoui) [inline com.meetfranz.Franz.appdata.xml]: You could add other url's like bugtracker, help, faq or donation

File: com.meetfranz.Franz.appdata.xml (hunk 1 above)
Comment (bilelmoussaoui) [reply com.meetfranz.Franz.appdata.xml]: You will also have to add 
`<launchable type="desktop-id">com.meetfranz.Franz.desktop</launchable>`

File: com.meetfranz.Franz.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            "name": "franz-desktop",
+            "buildsystem": "simple",
+            "build-commands": [
+                "ar x franz_*.deb",
+                "rm -f franz_*.deb",
+                "tar xf data.tar.xz",
+                "rm -f control.tar.gz data.tar.xz debian-binary",
+                "cp -r usr/* opt/* /app",
+                "chmod -R a-s,go+rX,go-w /app/Franz",
+                "sed -i 's|Exec=.*|Exec=franz %U|' /app/share/applications/franz.desktop",
```
Comment (bilelmoussaoui) [inline com.meetfranz.Franz.json]: You can use `desktop-file-utils` here

File: com.meetfranz.Franz.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "base-version": "18.08",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "18.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "franz",
+    "separate-locales": false,
+    "rename-desktop-file": "franz.desktop",
+    "rename-icon": "franz",
+    "finish-args": [
+        "--share=ipc",
```
Comment (bilelmoussaoui) [inline com.meetfranz.Franz.json]: It doesn't work under wayland?

File: com.meetfranz.Franz.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                "chmod -R a-s,go+rX,go-w /app/Franz",
+                "sed -i 's|Exec=.*|Exec=franz %U|' /app/share/applications/franz.desktop",
+                "install franz.sh /app/bin/franz",
+                "install -Dm644 com.meetfranz.Franz.appdata.xml /app/share/appdata/com.meetfranz.Franz.appdata.xml"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "only-arches": ["x86_64"],
+                    "url": "https://github.com/meetfranz/franz/releases/download/v5.0.0-beta.24/franz_5.0.0-beta.24_amd64.deb",
```
Comment (bilelmoussaoui) [inline com.meetfranz.Franz.json]: Flathub is about stable and reproducible builds. I don't have any idea if this beta version is considered as stable?

**Comment by bilelmoussaoui** [comment]
Have you contacted upstream about this? The appdata should be sent upstream as well.

File: com.meetfranz.Franz.appdata.xml (hunk 2 above)
Comment (vbatts) [reply com.meetfranz.Franz.appdata.xml]: instead of that homepage? or just multiple `<url/>`?

File: com.meetfranz.Franz.json (hunk 3 above)
Comment (vbatts) [reply com.meetfranz.Franz.json]: ah. this is carried over from `org.signal.Signal`

File: com.meetfranz.Franz.appdata.xml (hunk 2 above)
Comment (bilelmoussaoui) [reply com.meetfranz.Franz.appdata.xml]: multiple url's

File: com.meetfranz.Franz.json (hunk 3 above)
Comment (vbatts) [reply com.meetfranz.Franz.json]: example?

File: com.meetfranz.Franz.json (hunk 5 above)
Comment (vbatts) [reply com.meetfranz.Franz.json]: good question. I'm using it, but it is clearly marked as beta. I've sent a note to the author of the code.

File: com.meetfranz.Franz.json (hunk 4 above)
Comment (nedrichards) [reply com.meetfranz.Franz.json]: It's an electron app so I suspect not right now.

**Comment by nedrichards** [comment]
bot, build com.meetfranz.Franz
testbot, build com.meetfranz.Franz

**Comment by nedrichards** [comment]
Is there a reason why you can't build this app from source?

**Comment by vbatts** [comment]
> Is there a reason why you can't build this app from source?

Other than "I haven't", I do not see why not. Started from the example of org.signal.Signal which is an electron app from a *.deb

File: com.meetfranz.Franz.json (hunk 3 above)
Comment (bilelmoussaoui) [reply com.meetfranz.Franz.json]: https://github.com/flathub/com.github.z.Cumulonimbus/blob/785266a8ec0c2ebb73eea05571518238e193eb0c/com.github.z.Cumulonimbus.json#L32

**Comment by barthalion** [comment]
It's sometimes not worth it due to wonderful Electron/node ecosystem but we usually expect at least some attempts.

**Comment by nedrichards** [comment]
bot, build com.meetfranz.Franz

**Comment by vbatts** [comment]
An icon is too large? Hmm

-------- Original Message --------

**Comment by nedrichards** [comment]
@vbatts per spec 512x512 is the largest icon size you're allowed.

**Comment by nedrichards** [comment]
bot, build com.meetfranz.Franz

**Comment by nedrichards** [comment]
Closed in favour of https://github.com/flathub/flathub/pull/974 but with an inviration sent to @vbatts

**Pull request closed without merging**